src/libraries/Microsoft.Extensions.Logging/tests/Microsoft.Extensions.Logging.Generators.Tests/Microsoft.Extensions.Logging.Generators.Tests.csproj

```diff
@@ -0,0 +1,40 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
+    <EnableDefaultItems>true</EnableDefaultItems>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <MicrosoftNetCompilersToolsetVersion>3.10.0-1.final</MicrosoftNetCompilersToolsetVersion>
```

Comment:
Do we need to override this? Shouldn't we take it from the rest of the repo?

Reply:
This is needed to fix:

```
CSC : error CS8032: An instance of analyzer Microsoft.Extensions.Logging.Generators.LoggerMessageGenerator cannot be created from C:\CodeHub\runtime\artifacts\bin\Microsoft.Extensions.Logging.Generators\netstandard2.0-Release\Microsoft.Extensions.Logging.Generators.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=3.10.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.Logging\tests\Microsoft.Extensions.Logging.Generators.Tests\Microsoft.Extensions.Logging.Generators.Tests.csproj]
```

Reply:
Aren't users of the source generator going to have the same issue?

Reply:
Right: if they are using an older compiler.  I think a compatible version of the compiler is being used in latest build of the SDK https://github.com/dotnet/sdk/blob/3560c4d427bbb8ddccea25c329ccb493c839e12e/eng/Versions.props#L117-L121.  We should reference a minimum version that has the features we require, that way our nuget package works in the most possible scenarios.  I know the roslyn folks were discussing roslyn-version targeting conventions and how this is surfaced to the developer: https://github.com/dotnet/roslyn/pull/52423.

Reply:
I ended up reverting this change since my changes work does not need it
https://github.com/dotnet/runtime/pull/51064/commits/1b7a1583d3c77429305bf8427a8c74e6eed2e916

Reply:
I only upgraded so I could use the new `GeneratorPostInitializationContext` feature to support pre-generating the `JsonSerializableAttribute` (https://github.com/dotnet/runtimelab/pull/877) into user assemblies. For now, we are shipping the attribute from STJ.dll itself so we can stay on `3.8`. `3.9` is the minimum version needed to use the new feature.